### PR TITLE
backend/feat/auto-un-list

### PR DESCRIPTION
**Pull request by piyushKumar-1**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/DriverStats.hs [hunk 1]
```diff
@@ -23,6 +23,8 @@ data DriverStats = DriverStats
   { driverId :: Id Driver,
     idleSince :: UTCTime,
     totalRides :: Int,
-    totalDistance :: Meters
+    totalDistance :: Meters,
+    ridesCancelled :: Maybe Int,
+    totalRidesAssigned :: Maybe Int
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/DriverStats.hs:28]: Difference b/w totalRide and totalRidesAssigned?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/BookingCancellationReason.hs [hunk 2]
```diff
@@ -25,6 +26,22 @@ import Storage.Tabular.BookingCancellationReason
...
 create = Esq.create
 
+findAllCancelledByDriverId ::
+  Transactionable m =>
+  Id Person ->
+  m Int
+findAllCancelledByDriverId driverId = do
+  mkCount <$> do
+    Esq.findAll $ do
+      rideBookingCancellationReason <- from $ table @BookingCancellationReasonT
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/BookingCancellationReason.hs:36]: why finding in cacelation reason table? You should find in booking table?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/DriverStats.hs (hunk 1 above)
Comment (piyushKumar-1) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/DriverStats.hs:28]: Total ride is actually total completed rides for some reason 🙃. So can't use that.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/BookingCancellationReason.hs (hunk 2 above)
Comment (piyushKumar-1) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/BookingCancellationReason.hs:36]: Booking table doesn't have driver id will have to look in ride table , and to penalise a driver we will eventually have to fetch cancelled by driver from this table.

**Review by hkmangla — APPROVED** [review]

**Pull request merged**